lib/util.js

```diff
@@ -622,16 +622,24 @@ function formatObject(ctx, value, recurseTimes, visibleKeys, keys) {
 
 function formatArray(ctx, value, recurseTimes, visibleKeys, keys) {
   var output = [];
-  const maxLength = Math.min(Math.max(0, ctx.maxArrayLength), value.length);
-  const remaining = value.length - maxLength;
-  for (var i = 0; i < maxLength; ++i) {
-    if (hasOwnProperty(value, String(i))) {
-      output.push(formatProperty(ctx, value, recurseTimes, visibleKeys,
-                                 String(i), true));
+  let visibleLength = 0;
+  let index = 0;
+  while (index < value.length && visibleLength < ctx.maxArrayLength) {
+    let emptyElements = 0;
+    while (index < value.length && !hasOwnProperty(value, String(index))) {
+      emptyElements++;
+      index++;
+    }
+    if (emptyElements > 0) {
+      output.push(ctx.stylize(`[${emptyElements} empty]`, 'special'));
```

Comment:
Well, let's see wait and see what others think about it. If there's no specific preference, I'll change it to `'undefined'`.